src/Domain/Apprenticeship/Apprenticeship.cs

```diff
@@ -2,76 +2,84 @@
 using SFA.DAS.Funding.ApprenticeshipEarnings.Domain.Apprenticeship.Events;
 using SFA.DAS.Funding.ApprenticeshipEarnings.Domain.ApprenticeshipFunding;
 
-namespace SFA.DAS.Funding.ApprenticeshipEarnings.Domain.Apprenticeship
+namespace SFA.DAS.Funding.ApprenticeshipEarnings.Domain.Apprenticeship;
+
+public class Apprenticeship : AggregateRoot
 {
-    public class Apprenticeship : AggregateRoot
+    public Apprenticeship(Guid apprenticeshipKey, long approvalsApprenticeshipId, string uln, long ukprn, long employerAccountId, string legalEntityName, DateTime actualStartDate, DateTime plannedEndDate, decimal agreedPrice, string trainingCode, long? fundingEmployerAccountId, FundingType fundingType, decimal fundingBandMaximum, int ageAtStartOfApprenticeship)
     {
-        public Apprenticeship(Guid apprenticeshipKey, long approvalsApprenticeshipId, string uln, long ukprn, long employerAccountId, string legalEntityName, DateTime actualStartDate, DateTime plannedEndDate, decimal agreedPrice, string trainingCode, long? fundingEmployerAccountId, FundingType fundingType, decimal fundingBandMaximum, int ageAtStartOfApprenticeship)
-        {
-            ApprenticeshipKey = apprenticeshipKey;
-            ApprovalsApprenticeshipId = approvalsApprenticeshipId;
-            Uln = uln;
-            UKPRN = ukprn;
-            EmployerAccountId = employerAccountId;
-            LegalEntityName = legalEntityName;
-            ActualStartDate = actualStartDate;
-            PlannedEndDate = plannedEndDate;
-            AgreedPrice = agreedPrice;
-            TrainingCode = trainingCode;
-            FundingEmployerAccountId = fundingEmployerAccountId;
-            FundingType = fundingType;
-            FundingBandMaximum = fundingBandMaximum;
-            AgeAtStartOfApprenticeship = ageAtStartOfApprenticeship;
-        }
+        ApprenticeshipKey = apprenticeshipKey;
+        ApprovalsApprenticeshipId = approvalsApprenticeshipId;
+        Uln = uln;
+        UKPRN = ukprn;
+        EmployerAccountId = employerAccountId;
+        LegalEntityName = legalEntityName;
+        ActualStartDate = actualStartDate;
+        PlannedEndDate = plannedEndDate;
+        AgreedPrice = agreedPrice;
+        TrainingCode = trainingCode;
+        FundingEmployerAccountId = fundingEmployerAccountId;
+        FundingType = fundingType;
+        FundingBandMaximum = fundingBandMaximum;
+        AgeAtStartOfApprenticeship = ageAtStartOfApprenticeship;
+    }
+
+    public Apprenticeship(Guid apprenticeshipKey, long approvalsApprenticeshipId, string uln, long ukprn,
```

Comment:
I know most of the properties are 'get' only so we need to populate via a constructor, but this is quite a lot of parameters. Could we maybe construct this object in a different way?